lib/matplotlib/artist.py

```diff
@@ -1227,7 +1227,10 @@ def get_valid_values(self, attr):
         # Much faster than list(inspect.signature(func).parameters)[1],
         # although barely relevant wrt. matplotlib's total import time.
         param_name = func.__code__.co_varnames[1]
-        match = re.search("(?m)^ *{} : (.+)".format(param_name), docstring)
+        # We could set the number of * based on whether the parameter is a
+        # var(kw)args but it's not really worth the complexity.
+        match = re.search("(?m)^ *\*{{0,2}}{} : (.+)".format(param_name),
```

Comment:
why did this regexp change?

Reply:
The change is `\*{{0,2}}` to support parsing
~~~
    *margins : float, optional
~~~
and potentially also `**kwargs : type` (though I don't know if we have such a case. See also the added test.

Reply:
I just realized that this can never be `**kwargs` (because the setter passes the argument positionally), so I replaced it to only support a single `*`.